src/Tasks/Microsoft.NET.Build.Tasks/ResolveFrameworkReferences.cs

```diff
@@ -1,431 +1,62 @@
-                        packageToReference.SetMetadata(MetadataKeys.Version, items[1]);
-
-                        legacyFrameworkPackages.Add(packageToReference);
-                    }
-                }
-
-                //  Get the path of the targeting pack in the targeting pack root (e.g. dotnet/ref)
-                TaskItem targetingPack = new TaskItem(knownFrameworkReference.Name);
-                targetingPack.SetMetadata(MetadataKeys.PackageName, knownFrameworkReference.TargetingPackName);
-
-                string targetingPackVersion = null;
-                if (frameworkReference != null)
-                {
-                    //  Allow targeting pack version to be overridden via metadata on FrameworkReference
-                    targetingPackVersion = frameworkReference.GetMetadata("TargetingPackVersion");
-                }
-                if (string.IsNullOrEmpty(targetingPackVersion))
-                {
-                    targetingPackVersion = knownFrameworkReference.TargetingPackVersion;
-                }
-                targetingPack.SetMetadata(MetadataKeys.PackageVersion, targetingPackVersion);
-                targetingPack.SetMetadata("TargetingPackFormat", knownFrameworkReference.TargetingPackFormat);
-                targetingPack.SetMetadata("TargetFramework", knownFrameworkReference.TargetFramework.GetShortFolderName());
-                targetingPack.SetMetadata("RuntimeFrameworkName", knownFrameworkReference.RuntimeFrameworkName);
-                if (!string.IsNullOrEmpty(knownFrameworkReference.Profile))
-                {
-                    targetingPack.SetMetadata("Profile", knownFrameworkReference.Profile);
-                }
-
-                string targetingPackPath = null;
-                if (!string.IsNullOrEmpty(TargetingPackRoot))
-                {
-                    targetingPackPath = Path.Combine(TargetingPackRoot, knownFrameworkReference.TargetingPackName, targetingPackVersion);
-                }
-                if (targetingPackPath != null && Directory.Exists(targetingPackPath))
-                {
-                    // Use targeting pack from packs folder
-                    targetingPack.SetMetadata(MetadataKeys.PackageDirectory, targetingPackPath);
-                    targetingPack.SetMetadata(MetadataKeys.Path, targetingPackPath);
-                }
-                else
-                {
-                    if (EnableTargetingPackDownload)
-                    {
-                        //  Download targeting pack
-                        TaskItem packageToDownload = new TaskItem(knownFrameworkReference.TargetingPackName);
-                        packageToDownload.SetMetadata(MetadataKeys.Version, targetingPackVersion);
-
-                        packagesToDownload.Add(packageToDownload);
-                    }
-                }
+                TaskItem resolvedFrameworkReference = new TaskItem(frameworkReference.ItemSpec);
```

Comment:
We're expecting metadata (including IsImplicitlyDefined/PrivateAssets) to be copied from the original item -> resolved item, I think (?) this is what ResolveAssemblyReferences does.

Reply:
I would expect IsImplicitlyDefined to determine whether you can delete the reference from the solution tree, but what would you do with PrivateAssets?

Reply:
It shows up in the Properties page so that you can flip it.

Reply:
I'm not sure I consider PrivateAssets on FrameworkReference items a user-facing feature.  We use it to communicate that an implicit FrameworkReference is represented by the target framework, and thus doesn't need to flow transitively.  I don't think there's a scenario for setting it on other FrameworkReferences, and if it shows up in the UI it's more likely to cause confusion and incorrect builds if people set it.

Reply:
You might have a reference to a project and not want to inherit its FrameworkReference.